var/spack/repos/builtin/packages/py-msrest/package.py

```diff
@@ -13,6 +13,16 @@ class PyMsrest(PythonPackage):
     homepage = "https://github.com/Azure/msrest-for-python"
     pypi = "msrest/msrest-0.6.16.tar.gz"
 
+    # There is some issue with recent Azure packages. They are not downloaded successfully
+    # by Spack if URL is not provided. Maybe due to `.zip` format of the archive.
+    # `spack checksum` command also no longer sees new packages.
```

Comment:
My recommendation would be to update `pypi` to `.zip` so that new versions can be added and add a `url_for_version` that switches from `.tar.gz` to `.zip` based on version.

Reply:
Replaced with a procedure.